client/my-sites/plugins/plugin-site-list/index.jsx

```diff
@@ -1,59 +1,52 @@
 /**
  * External dependencies
  */
+import classNames from 'classnames';
 import { compact } from 'lodash';
-var React = require( 'react' ),
-	classNames = require( 'classnames' );
+import { connect } from 'react-redux';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
-var allSites = require( 'lib/sites-list' )(),
-	PluginSite = require( 'my-sites/plugins/plugin-site/plugin-site' ),
-	SectionHeader = require( 'components/section-header' ),
-	PluginsStore = require( 'lib/plugins/store' );
+import { isConnectedSecondaryNetworkSite, getNetworkSites } from 'state/selectors';
+import PluginSite from 'my-sites/plugins/plugin-site/plugin-site';
+import PluginsStore from 'lib/plugins/store';
+import SectionHeader from 'components/section-header';
 
-module.exports = React.createClass( {
+export class PluginSiteList extends Component {
+	static propTypes = {
+		notices: PropTypes.object,
+		plugin: PropTypes.object,
+		sites: PropTypes.array,
+		sitesWithSecondarySites: PropTypes.array,
+		title: PropTypes.string,
+	};
 
-	displayName: 'PluginSiteList',
-
-	propTypes: {
-		site: React.PropTypes.object,
-		plugin: React.PropTypes.object,
-		notices: React.PropTypes.object,
-		title: React.PropTypes.string
-	},
-
-	getSecondaryPluginSites: function( site ) {
-		let secondarySites = allSites.getNetworkSites( site );
-		let secondaryPluginSites = site.plugin
+	getSecondaryPluginSites( site, secondarySites ) {
+		const secondaryPluginSites = site.plugin
 			? PluginsStore.getSites( secondarySites, this.props.plugin.slug )
 			: secondarySites;
 		return compact( secondaryPluginSites );
-	},
+	}
 
-	renderPluginSite: function( site ) {
+	renderPluginSite( { site, secondarySites } ) {
 		return <PluginSite
 				key={ 'pluginSite' + site.ID }
 				site={ site }
-				secondarySites={ this.getSecondaryPluginSites( site ) }
+				secondarySites={ this.getSecondaryPluginSites( secondarySites ) }
```

Comment:
`getSecondaryPluginSites` should receive both `site` and `secondarySites` as params here.